CAN_Logger_with_Autobaud_and_Requests/CAN_Logger_with_Autobaud_and_Requests.ino

```diff
@@ -0,0 +1,1174 @@
+  }
+  
+  // Send requests for additional data if needed
+  if (send_requests){
+    if (send_passes < NUM_REQUEST_PASSES){
+      if (send_request_timer > REQUEST_TIMING){
+        send_request_timer = 0;
+        txmsg.len = 3;
+        txmsg.id = 0x18EA00F9;
+        txmsg.buf[0] = (request_pgn[request_index] & 0x0000FF);
+        txmsg.buf[1] = (request_pgn[request_index] & 0x00FF00) >> 8 ;
+        txmsg.buf[2] = (request_pgn[request_index] & 0xFF0000) >> 16; //These are in reverse byte order.
+        send_Can0_message(txmsg);
+        if (RXCount1 > 0) send_Can1_message(txmsg);
+        if (RXCount2 > 0) send_Can2_message(txmsg);
+        //Toggle LED light as messages are sent
+        RED_LED_state = !RED_LED_state;                       
+        digitalWrite(RED_LED,RED_LED_state);  
+        request_index++;
+        if (request_index >= NUM_REQUESTS) {
+          request_index = 0;
+          send_passes++;
+          //shuffle
+          //Serial.println("Shuffling Requests");
+          for (int i = 0; i < NUM_REQUESTS; i++) {
+            int j = random(i, NUM_REQUESTS);
+            auto temp = request_pgn[i];
+            request_pgn[i] = request_pgn[j];
+            request_pgn[j] = temp;
+          }
+        }
+      }
+    }
+    else
+    {
+      send_passes = 0;
+      send_requests = false;
+      send_iso_requests = true;
+    }
+  }
+  else if (send_iso_requests)
+  {
+    if (send_iso_passes < NUM_REQUEST_PASSES){
+      if (send_iso_request_timer > REQUEST_TIMING){
+        send_iso_request_timer = 0;
+        txmsg.len = 8;
+        txmsg.id = 0x18DA00F9;
+        txmsg.buf[0] = 0x02;
+        txmsg.buf[1] = (iso_request[request_index] & 0x00FF) >> 0 ;
+        txmsg.buf[2] = (iso_request[request_index] & 0xFF00) >> 8; //These are in reverse byte order.
+        send_Can0_message(txmsg);
+        if (RXCount1 > 0) send_Can1_message(txmsg); //onlt send something if the network is alive
+        if (RXCount2 > 0) send_Can2_message(txmsg);
+        //Toggle LED light as messages are sent
+        RED_LED_state = !RED_LED_state;                       
+        digitalWrite(RED_LED,RED_LED_state);  
+        request_index++;
+        if (request_index >= NUM_REQUESTS) {
+          request_index = 0;
+          send_iso_passes++;
+          //shuffle
+          //Serial.println("Shuffling Requests");
+          for (int i = 0; i < NUM_ISO_REQUESTS; i++) {
+            int j = random(i, NUM_ISO_REQUESTS);
+            auto temp = iso_request[i];
+            iso_request[i] = request_pgn[j];
+            iso_request[j] = temp;
+          }
+        }
+      }
+    }
+    else
+    {
+      send_passes = 0;
+      send_iso_requests = false;
+    }
+  }
+
+  // Blink the LEDs if needed
+  led_blink_routines();
+
+  //digitalWrite(RED_LED,digitalRead(POWER_PIN));
+  
+  if (Serial.available() >= 2) {
+    commandString = Serial.readStringUntil('\n');
+    if (commandString.equalsIgnoreCase("HEX"))          print_hex();
```

Comment:
if you're going to ship a serial console, you should include a HELP command that lists the available commands.

Reply:
@duyvan1995 will add this feature.